### PR TITLE
Refined 'tests/array_split.py' w/ more essential input shapes

### PR DESCRIPTION
**Pull request by sbak5**
Refined this test w/ reduced input shapes. 
Instead, more essential test cases are included w/ empty, singleton arrays

### CODE REVIEW

File: tests/array_split.py [hunk 1]
```diff
@@ -22,127 +22,110 @@
...
-                                even_div,
-                            )
-                        )
+
+    input_shapes = [(0), (1), (1, 1), (1, 1, 1), (dim, dim), (dim, dim, dim)]
+    for shape in input_shapes:
+        a = np.random.randint(low=0, high=100, size=shape)
+        for axis in range(a.ndim):
+            input_arr = []
+            even_div = None  # a.shape[axis]
```
Comment (magnatelee) [inline tests/array_split.py:39]: what does the inline comment mean?

File: tests/array_split.py [hunk 2]
```diff
@@ -22,127 +22,110 @@
...
-                        list(
-                            range(
-                                0,
-                                a.shape[axis]
-                                + even_div * np.random.randint(1, 10),
-                                even_div,
-                            )
-                        )
+
+    input_shapes = [(0), (1), (1, 1), (1, 1, 1), (dim, dim), (dim, dim, dim)]
```
Comment (magnatelee) [inline tests/array_split.py:34]: I guess you could also add empty arrays with more convoluted shapes, like (0, 10).

File: tests/array_split.py [hunk 3]
```diff
@@ -22,127 +22,110 @@
...
+                else:
+                    uneven_div = div
+                if even_div is not None and uneven_div is not None:
+                    break
+            if even_div is None:
+                even_div = a.shape[axis] if (a.shape[axis] >= 1) else 1
+            # divisible integer
+            input_arr.append(even_div)
+            # indivisble integer
+            if even_div != uneven_div and uneven_div is not None:
```
Comment (magnatelee) [inline tests/array_split.py:53]: the first condition seems very confusing. I believe `even_div` and `uneven_div` meant to be different, but why are you checking if they are different?

File: tests/array_split.py [hunk 4]
```diff
@@ -22,127 +22,110 @@
...
+        for axis in range(a.ndim):
+            input_arr = []
+            even_div = None  # a.shape[axis]
+            uneven_div = None  # a.shape[axis]
+            for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
+                if a.shape[axis] % div == 0:
+                    even_div = div
+                else:
+                    uneven_div = div
+                if even_div is not None and uneven_div is not None:
```
Comment (magnatelee) [inline tests/array_split.py:46]: I don't think I'm a big fan of this condition, as the values that would be chosen for `even_div` and `uneven_div` are somewhat unpredictable. I feel we're trying to be too smart here, as we won't really run this test with non-default test parameters.

File: tests/array_split.py [hunk 5]
```diff
@@ -22,127 +22,110 @@
...
-                        print_msg = (
-                            f"np.{routine}({a.shape}, {input_opt}" f", {axis})"
-                        )
-                        # Check if both impls produce the error
-                        # for non-viable options
-                        b = getattr(np, routine)(a, input_opt, axis)
-                        c = getattr(num, routine)(a, input_opt, axis)
-                        is_equal = True
-                        err_arr = [b, c]
+            for routine, input_opt in itertools.product(
```
Comment (magnatelee) [inline tests/array_split.py:82]: again, what's the point of doing `product` if this test only tests `array_split`?

File: tests/array_split.py [hunk 6]
```diff
@@ -22,127 +22,110 @@
...
+                is_equal = True
+                err_arr = [b, c]
 
-                        if len(b) != len(c):
+                if len(b) != len(c):
+                    is_equal = False
+                    err_arr = [b, c]
+                else:
+                    for each in zip(b, c):
+                        sub_set = list(each)
```
Comment (magnatelee) [inline tests/array_split.py:99]: why do you need to create a fresh list here? isn't each already a tuple that you can access with index expression?

File: tests/array_split.py (hunk 3 above)
Comment (sbak5) [reply tests/array_split.py:53]: It can both 1 if a.shape[axis] is prime.

File: tests/array_split.py (hunk 5 above)
Comment (sbak5) [reply tests/array_split.py:82]: It used to have `split` w/ error checking codes to see if both cunumeric and numpy produces the same error for non-viable options. I'll delete this

File: tests/array_split.py (hunk 4 above)
Comment (sbak5) [reply tests/array_split.py:46]: I see. This tests take any size of matrix so the dimension could be prime or not, which I tried to handle that. 
What do you mean by 'non-defualt' test parameters?

Comment (magnatelee) [reply tests/array_split.py:53]: I mean, practically no one will try to run this unit test with the size overriden by the `--dim_size` flag. I'm saying the test is overly complicated for cases that would happen very rarely.

File: tests/array_split.py (hunk 3 above)
Comment (magnatelee) [reply tests/array_split.py:60]: ok then is there reason that you don't test with `even_div == a.shape[axis]` in that case? I feel this is another unnecessary complication in this test that is just confusing.

File: tests/array_split.py [hunk 7]
```diff
@@ -13,136 +13,115 @@
...
-                        b = getattr(np, routine)(a, input_opt, axis)
-                        c = getattr(num, routine)(a, input_opt, axis)
-                        is_equal = True
-                        err_arr = [b, c]
+            for input_opt in input_arr:
+                # test divisible integer or indices
+                print_msg = f"np.{routine}({a.shape}, {input_opt}" f", {axis})"
+                # Check if both impls produce the error
+                # for non-viable options
+                b = getattr(np, routine)(a, input_opt, axis)
```
Comment (magnatelee) [inline tests/array_split.py:96]: again, why can't you just call `np.array_split` here?

**Review by magnatelee — COMMENTED** [review]
I only have one petty comment, looks good otherwise.

**Comment by magnatelee** [comment]
Looks good to me.
